src/browser/ui/ReactMount.js

```diff
@@ -451,12 +458,27 @@ var ReactMount = {
    * perform an update on it and only mutate the DOM as necessary to reflect the
    * latest React component.
    *
+   * @param {ReactComponent} parentComponent The conceptual parent of this render tree.
    * @param {ReactElement} nextElement Component element to render.
    * @param {DOMElement} container DOM element to render into.
    * @param {?function} callback function triggered on completion
    * @return {ReactComponent} Component instance rendered in `container`.
    */
-  render: function(nextElement, container, callback) {
+  renderSubtreeIntoContainer: function(parentComponent, nextElement, container, callback) {
+    console.log(typeof parentComponent._reactInternalInstance);
+    invariant(
+      parentComponent != null && parentComponent._reactInternalInstance != null,
+      'parentComponent must be a valid React Component'
+    );
+    invariant(
+      ReactContext.inContextLifecycle,
+      'Must be in a lifecycle method to invoke renderSubtreeIntoContainer'
```

Comment:
@sebmarkbage said, for now, only componentDidMount and componentDidUpdate are allowed.  Currently calling it from a callback is not allowed (although I don't like this, I think it should be allowed, it's always easy to remove this restriction later).  Calling it twice in a row shouldn't be a problem, though I can't imagine why someone would want to do that.  Calling it from within Render is (according to sebastian) not good and disallowed.  I'll clarify the error message.